src/lib.rs

```diff
@@ -416,3 +417,9 @@ impl Inner {
         true
     }
 }
+
+impl Wake for Inner {
+    fn wake(self: Arc<Self>) {
+        self.unpark();
+    }
+}
```

Comment:
Can you also implement [`wake_by_ref`](https://doc.rust-lang.org/std/task/trait.Wake.html#method.wake_by_ref)? It would optimize certain cases where ownership of the waker is not needed.

Reply:
It would be the same right, there will be no differences I assume?